### PR TITLE
[For review only] Hole prototype

### PR DESCRIPTION
**Pull request by mravn-google**
Prototype of "punch a hole through the Flutter UI" to let other platform views show.
Cf. https://github.com/flutter/engine/pull/4391

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Holes are only needed on Android, correct?

**Comment by mravn-google** [comment]
I don't know. I haven't looked at iOS at all.

File: packages/flutter/lib/src/rendering/texture.dart [hunk 1]
```diff
@@ -78,3 +78,33 @@ class TextureBox extends RenderBox {
...
+
+  @override
+  bool hitTestSelf(Offset position) {
+    return true;
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    context.addLayer(new HoleLayer(
+      rect: new Rect.fromLTWH(offset.dx, offset.dy, size.width, size.height)
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/texture.dart]: you can just say `rect: offset & size`

**Comment by Hixie** [comment]
(This would documentation and tests before checkin.)

See questions on flutter/engine#4391.

**Pull request closed without merging**